### ISSUE TITLE
[Feature Request] CreateProcess "No WaitForExit option"

### ISSUE THREAD

**Issue by shutingrz** [cobbr/SharpSploit#59]
For Covenant's Grunt redundancy, I want to do the same thing as "spawn" as implemented in PowerShell Empire.

However, SharpSploit's Execute calls WaitForExit(), so control won't return until the newly started process is finished.

```
process.WaitForExit();
return "";
```

Could you add an option for the Execute and CreateProcess methods to not call WaitForExit()?

**Comment by rasta-mouse** [comment]
I think there are better ways to achieve a `spawn` Task in Covenant, personally.

**Comment by shutingrz** [comment]
Thanks for the reply.

I'm not sure if it's right to continue talking about Covenant here, but I'll talk about Coevnant.

GruntSMB running on C3's Relay has an issue where Grunt will not respond until the Task is completely finished.
In other words, the behavior is synchronous, meaning that if you can't complete a single Task, it stops immediately.
We should be able to identify the problem and fix it, but this is time consuming.

That's why I wanted to launch more than one Grunt(Relay) as a WorkAround in case Grunt stops responding.

Fortunately, Covenant allows you to create custom Tasks, so I changed the Shell Task and created a new Shell Task that does not use WaitForExit(). Thus the problem is temporarily solved.

However, for me personally, there have been times when I've wanted to start a new process without waiting for a process to end, not just "spawn".
If this idea is in the minority, I would close Issue.

**Comment by checkymander** [comment]
I don't necessarily think giving the operators the ability to kick off an executable and move on is the worst idea. Would be pretty easy to implement I think, we could just add an optional overload to the Execute() line of tasks called WaitForExit or whatever and then change the process launch code to

```
if(WaitForExit){
    process.StartInfo.WaitForExit();
    //read output
}
else{
    return "Process Executed"
}
```

**Comment by rasta-mouse** [comment]
Perhaps.  Or would you just thread it in your application instead and have a callback?

**Comment by shutingrz** [comment]
I've already created a Shell Task without WaitForExit() in my custom Task, and so I think it's technically simple.
However, the original Shell Task used SharpSploit, so I wanted to respect that design and created an Issue in the SharpSploit repository.

However, I realized that, in general, it was preferable to change Covenant's Task instead of changing the code in SharpSploit.
(Because it's a simple request and I think there's little demand for it)

For that reason, I'm going to close this Issue.
